bin/kmer_count_dim_reduction.py

```diff
@@ -0,0 +1,310 @@
+        embedding_title = "KernelPCA"
+        print_timestamp(embedding_title)
+        kernel_pca = decomposition.KernelPCA(n_components=2, kernel="sigmoid")
+        embedding = kernel_pca.fit_transform(df)
+
+    elif selected_method == "pca_svd":
+        #https://www.tutorialspoint.com/scikit_learn/scikit_learn_dimensionality_reduction_using_pca.htm
+        embedding_title = "PCA with SVD solver"
+        print_timestamp(embedding_title)
+        pca = decomposition.PCA(n_components=2, svd_solver="randomized")
+        embedding = pca.fit_transform(df)
+
+
+    elif selected_method == "autoencoder_sigmoid":
+        print_timestamp("Autoencoder sigmoid")
+        #https://ekamperi.github.io/machine%20learning/2021/01/21/encoder-decoder-model.html
+        embedding = run_autoencoder(df, autoencoder_epochs_count, "sigmoid", out_folder)
+
+    elif selected_method == "autoencoder_linear":
+        #https://ekamperi.github.io/machine%20learning/2021/01/21/encoder-decoder-model.html
+        embedding_title = "Autoencoder linear"
+        print_timestamp(embedding_title)
+        embedding = run_autoencoder(df, autoencoder_epochs_count, "linear", out_folder)
+
+    elif selected_method == "autoencoder_tanh":
+        #https://ekamperi.github.io/machine%20learning/2021/01/21/encoder-decoder-model.html
+        embedding_title = "Autoencoder tanh"
+        print_timestamp(embedding_title)
+        embedding = run_autoencoder(df, autoencoder_epochs_count, "tanh", out_folder)
+
+    elif selected_method == "autoencoder_selu":
+        #https://ekamperi.github.io/machine%20learning/2021/01/21/encoder-decoder-model.html
+        embedding_title = "Autoencoder selu"
+        print_timestamp(embedding_title)
+        embedding = run_autoencoder(df, autoencoder_epochs_count, "selu", out_folder)
+
+    elif selected_method == "autoencoder_relu":
+        #https://ekamperi.github.io/machine%20learning/2021/01/21/encoder-decoder-model.html
+        embedding_title = "Autoencoder relu"
+        print_timestamp(embedding_title)
+        embedding = run_autoencoder(df, autoencoder_epochs_count, "relu", out_folder)
+
+
+    elif selected_method == "nmf":
+        #https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.NMF.html
+        embedding_title = "Non-Negative Matrix Factorization"
+        print_timestamp(embedding_title)
+        nmf_model = decomposition.NMF(n_components=2, init="random", random_state=0)
+        embedding = nmf_model.fit_transform(df)
+    return embedding, embedding_title
+
+
+def main(kmer_counts_file, out_folder, selected_methods, n_neighbors_setting, autoencoder_epochs_count):
+
+    reset_random_seeds()
+    
+    df, seq_names = load_data(kmer_counts_file)
+    df_row_count = df.shape[0]
+    if df_row_count == 1:
+        sys.stderr.write("Skipping the dimensionality reduction of kmer counts, as the kmer counts table has only one row")
```

Comment:
I got a crash of the `kmer_count_dim_reduction.nf` module when running the subworkflow with an input FASTA file that has only one sequence (the error message is below). I think what happens is that the Python script exits without a crash like it should (exit code 0) but then the Nextflow script expects a file  `*_kmers_dim_reduction_embeddings.csv` to be produced by the Python script but the file isn't there (because the Python script skipped making it)

```
Execution cancelled -- Finishing pending tasks before exit
-[sanger-tol/ascc] Pipeline completed with errors-
ERROR ~ Error executing process > 'SANGERTOL_ASCC:ASCC:GET_KMERS_PROFILE:KMER_COUNT_DIM_REDUCTION (asccTinyTest)'                          

Caused by:                                                                                                                                 
  Missing output file(s) `*_kmers_dim_reduction_embeddings.csv` expected by process `SANGERTOL_ASCC:ASCC:GET_KMERS_PROFILE:KMER_COUNT_DIM_REDUCTION (asccTinyTest)` 
```